### PR TITLE
Fix config file parsing

### LINKED ISSUE #70: Couple of issues when parsing the config file
Running swayidle https://github.com/swaywm/swayidle/commit/04a176ee91aee6cca45aa91b95f2626ecb2fbb28

### Exhibit # 1: Config file without empty newline segfaults

```
$ cat -p swayidle/.config/swayidle/config
timeout 3 'swaylock'
before-sleep 'swaylock'
$ swayidle 
Segmentation fault (core dumped)
```

With an empty newline no problems
```
$ cat -p swayidle/.config/swayidle/config
timeout 3 'swaylock'
before-sleep 'swaylock'

$ swayidle 
Failed to find session
```

### Exhibit # 2: Command without quotes is not correctly parsed
```
$ cat -p swayidle/.config/swayidle/config
timeout 3 'swaylock'
timeout 6 'swaymsg "output * dpms off"' resume 'swaymsg "output * dpms on"'
before-sleep swaylock

$ swayidle -d
...
Command: swayloc
Setup sleep lock: swayloc
...
```

With quotes or a random char (in the example a dash `-`) fixes
```
$ cat -p swayidle/.config/swayidle/config
timeout 3 'swaylock'
timeout 6 'swaymsg "output * dpms off"' resume 'swaymsg "output * dpms on"'
before-sleep swaylock-

$ swayidle -d
...
Command: swaylock
Setup sleep lock: swaylock
...
```

### Exhibit # 3: Misc

* `before-sleep` directive without quotes refused
* UTF-8 is not supported (minor issue, just for completeness)
    `timeout 3 'swaylock🐶🔥'` => `Command: swaylock🐶�`

### PR DESCRIPTION
**Pull request by apiraino**
Hi!

Would this patch fix #70?

thanks for the review (and sorry for the perhaps poor quality of my patch)

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -927,7 +927,8 @@ static int load_config(const char *config_path) {
 	ssize_t nread;
 	while ((nread = getline(&line, &n, f)) != -1) {
 		lineno++;
-		line[nread-1] = '\0';
+                if (line[nread-1] == '\n')
```
Comment (emersion) [inline main.c:930]: Style: we use tabs, not spaces

Comment (emersion) [inline main.c:930]: Style: braces are mandatory

**Review by emersion — CHANGES_REQUESTED** [review]
Thanks for sending a patch, the logic LGTM!

Can you edit your commit message to include more information? Maybe something among these lines:

```
Fix last character of config file dropped

getline returns a string without a newline if the config file doesn't have a final newline.

Closes: #70 
```

File: main.c (hunk 1 above)
Comment (apiraino) [reply main.c]: the github editor caught me offguard :)

**Comment by emersion** [comment]
Can you squash these two commits together? See https://git-rebase.io/

**Comment by apiraino** [comment]
> Can you squash these two commits together? See https://git-rebase.io/

yes I can ... (I thought you could squash+merge directly from github)

**Comment by apiraino** [comment]
superseded by #87

**Comment by emersion** [comment]
There's no need to close the PR. You can force-push to your branch instead.

**Pull request closed without merging**